### PR TITLE
Feature/update go modules

### PR DESCRIPTION
**Pull request by kingcdavid**
### Changed
- Updated all go dependancies
- Stubbed TestCase and TestStep from github.com/hashicorp/vault/helper/testhelpers/logical to avoid loading vault as a module
- Updated action to build with golang version 1.21

### CODE REVIEW

File: path_roles_test.go [hunk 1]
```diff
@@ -28,12 +28,11 @@ import (
 
 	"github.com/fatih/structs"
 	"github.com/go-test/deep"
-	"github.com/hashicorp/errwrap"
 	"github.com/hashicorp/vault/sdk/helper/certutil"
 	"github.com/hashicorp/vault/sdk/helper/strutil"
 	"github.com/hashicorp/vault/sdk/logical"
 
-	logicaltest "github.com/hashicorp/vault/helper/testhelpers/logical"
+	//logicaltest "github.com/hashicorp/vault/helper/testhelpers/logical"
```
Comment (abhishekmishra0224) [inline path_roles_test.go]: Can we please remove the commented code.

Comment (kingcdavid) [reply path_roles_test.go]: removed

**Comment by abhishekmishra0224** [comment]
LGTM

**Review by abhishekmishra0224 — APPROVED** [review]
LGTM

**Review by abhishekmishra0224 — APPROVED** [review]

**Pull request merged**